### PR TITLE
Change bounds separator

### LINKED ISSUE #485: Can not share/copy the url of online map to social media
Please note, this is on the `master` branch.

https://dadior.s3.ap-northeast-1.amazonaws.com/Treetracker+by+Greenstand+-+Google+Chrome+2022-02-25+12-56-13.mp4

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by iib0011**
# Description

We use '!' instead of '," as a separator for the bounds. This is a way to get around Facebook's replacement from "," to "%" in query Parameters

Fixes #485

- [ ] Bug fix (non-breaking change which fixes an issue)

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

# How Has This Been Tested?

- [ ] Jest unit tests

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by dadiorchen** [comment]
@iib0011 can you solve the conflicts above? So out CI can get exectued.

**Comment by dadiorchen** [comment]
@pratikmdhr can you review this

**Comment by pratikmdhr** [comment]
@iib0011 @dadiorchen i think there is some issue with the merge, i don't see any bounds in the url.. 

https://user-images.githubusercontent.com/52301822/156964506-f9a59774-0d26-4b6a-a78a-b9ac9e9b94b4.mp4

.

File: src/App.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ import Logo from "./components/Logo";
 import SidePanel from "./components/SidePanel";
 import Timeline from "./components/Timeline";
 import getLogoSrc from "./models/logo";
-import Map from "./models/Map";
+import {Map} from "treetracker-web-map-core";
```
Comment (pratikmdhr) [inline src/App.js:21]: @dadiorchen @iib0011 we need to retain the Map from the master branch right??

File: src/App.js [hunk 2]
```diff
@@ -388,12 +388,15 @@ function App() {
...
     const parameters = getParameters();
+    parameters.bounds = parameters.bounds?.replaceAll('!',',');
     const map = new Map({
       onLoad: loaded,
       onClickTree: showPanel,
       onFindNearestAt: handleFindNearestAt,
       onError: handleError,
       filters: parameters,
+      tileServerUrl: process.env.REACT_APP_TILE_SERVER_URL,
+      tileServerSubdomains: process.env.REACT_APP_TILE_SERVER_SUBDOMAINS.split(","),
```
Comment (pratikmdhr) [inline src/App.js:399]: if the Map from the master branch is kept unchanged, this is not required

**Review by pratikmdhr — CHANGES_REQUESTED** [review]
@iib0011 @dadiorchen I will check the code after we fix the issue with the bounds.. until then i notice these issues with the merge

File: src/App.js (hunk 1 above)
Comment (iib0011) [reply src/App.js]: I changed it

File: src/App.js (hunk 2 above)
Comment (iib0011) [reply src/App.js]: done

**Comment by dadiorchen** [comment]
Oh it’s possible with the master branch. We probably need to patch a special version rather than master directly. I’m unavailable today and will be back tomorrow and I will help on this when I back

**Comment by dadiorchen** [comment]
Oh maybe I need to reverse the master in some way

File: src/models/Map.js [hunk 3]
```diff
@@ -724,7 +724,7 @@ export default class Map{
 
   updateUrl(){
     log.warn("update url");
-    window.history.pushState('treetrakcer', '', `/?${this.getFilterParameters()}&bounds=${this.getCurrentBounds()}`);
+    window.history.pushState('treetracker', '', `/?${this.getFilterParameters()}&bounds=${this.getCurrentBounds().replaceAll(',','!')}`);
```
Comment (pratikmdhr) [inline src/models/Map.js:727]: looks good

File: src/App.js [hunk 4]
```diff
@@ -388,6 +388,7 @@ function App() {
     script.id = 'googleMaps';
     document.body.appendChild(script);
     const parameters = getParameters();
+    parameters.bounds = parameters.bounds?.replaceAll('!',',');
```
Comment (pratikmdhr) [inline src/App.js:391]: looks good

File: package.json [hunk 5]
```diff
@@ -19,7 +19,8 @@
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-scripts": "3.4.1",
-    "source-map-explorer": "^2.5.0"
+    "source-map-explorer": "^2.5.0",
+    "treetracker-web-map-core": "^1.1.0"
```
Comment (pratikmdhr) [inline package.json:23]: @iib0011 i forgot to mark this one yesterday, you can delete this map-core package too

**Review by pratikmdhr — CHANGES_REQUESTED** [review]

**Comment by pratikmdhr** [comment]
hey @iib0011, your code to change the bound separator looks good to me! @dadiorchen the bounds appeared back now after @iib0011 's last commit

**Comment by dadiorchen** [comment]
We have soled this problem so I will close this PR now

**Pull request closed without merging**